Sources/Build/describe().swift

```diff
@@ -30,7 +30,7 @@ public func describe(_ prefix: AbsolutePath, _ conf: Configuration, _ modules: [
     let Xcc = flags.cCompilerFlags.flatMap{ ["-Xcc", $0] }
     let Xld = flags.linkerFlags.flatMap{ ["-Xlinker", $0] }
     let prefix = prefix.appending(component: conf.dirname)
-    try Utility.makeDirectories(prefix.asString)
+    try mkdir(prefix)
```

Comment:
I agree about the names-but-not-semantics argument, but I really want to avoid mixing all-lowercase short forms vs camel-case long forms.  So I would suggest either:
a) renaming all of these functions to names that can't be mistaken for POSIX names
or
b) changing the default of this optional parameter (and those of any other functions with POSIX names) so that the default semantics match POSIX, but that you can pass, for example, `recursive: true` as enhanced behavior.

Do you have a preference?